code/game/atoms.dm

```diff
@@ -743,9 +743,27 @@
 /atom/proc/get_color()
 	return color
 
-/// Set the color of this atom to `new_color`.
-/atom/proc/set_color(new_color)
-	color = new_color
+/* Set the atom colour. This is a stub effectively due to the broad use of direct setting. */
+// TODO: implement this everywhere that it should be used instead of direct setting.
+/atom/proc/set_color(var/new_color)
+	if(isnull(new_color))
+		return reset_color()
+	if(color != new_color)
+		color = new_color
+		return TRUE
+	return FALSE
+
+/atom/proc/reset_color()
+	if(color != COLOR_WHITE)
```

Comment:
i am pretty sure that setting color to `COLOR_WHITE` will set color to null actually, so it might assign it unnecessarily if color is already null. not a big deal though